explainers/prioritized-task-scheduling.md

```diff
@@ -0,0 +1,856 @@
+simpler for the use case, more ergonomic for async code, and supports inheritance as well.
+
+#### Future Enhancements
+
+An extension of this API would be to enable waiting for things other than time, e.g. events. We plan
+to explore integrating this with [observables](https://github.com/WICG/observable), depending on how
+that work proceeds.
+
+### `scheduler.currentTaskSignal`
+
+`scheduler.currentTaskSignal` returns the current task's signal, which is the `TaskSignal` used by
+the "yieldy APIs" (`scheduler.yield()` et al.) for signal inheritance. Exposing this signal enables
+using it to schedule related work or combining it with other signals without needing to pass the
+signal through every function along the way.
+
+**Example: reading the current task's priority.**
+
+```js
+function task() {
+  console.log(scheduler.currentTaskSignal.priority);  // 'background'
+}
+
+scheduler.postTask(task, {priority: 'background'});
+```
+
+**Example: combining signals with the current task's signal.**
+
+
+```js
+async function task() {
+  // Subtasks should be aborted if this task's signal is aborted, but can
+  // separately be aborted by this controller.
+  const controller = new AbortController();
+  const signal = TaskSignal.any(
+   [controller.signal, scheduler.currentTaskSignal],
+   {priority: scheduler.currentTaskSignal});
+
+  scheduler.postTask(subtask1, {signal});
+  scheduler.postTask(subtask2, {signal});
+
+  // ...
+}
+
+scheduler.postTask(task, {signal: someSignal});
+```
+
+### Integration with Other APIs
+
+
+
+UAs can prioritize on a per-[task
+source](https://html.spec.whatwg.org/multipage/webappapis.html#task-source) basis; but such
+decisions are difficult without explicit (userland) priority information. Integrating `scheduler`
+[priorities](#task-priorities) with other asynchronous APIs would give developers further control
+over task ordering on their pages by creating a more complete prioritized task scheduling system.
+
+#### `fetch`
+
+`fetch()` has a `signal` option (an `AbortSignal`) which is used to abort an ongoing fetch, similar
+to the `signal` option in `scheduler` APIs. We propose extending this such that if the signal
+provided is a `TaskSignal`, the priority is used for event loop task scheduling.
+
+**Example: A 'user-blocking' fetch.**
+```js
+async function task() {
+  // Both of the promises below are resolved in 'user-blocking' tasks.
+  const prioritySignal = TaskSignal.any([], {priority: 'user-blocking'});
+  let response = await fetch(url, {signal: prioritySignal});
+  let data = await response.json();
+}
+```
+
+This can be used to increase the priority such that fetch-related tasks won't be blocked by
+`'user-visible'` (default) continuations or tasks, or can be used to deprioritize fetches related to
+background work.
+
+Note that:
+ 1. This is separate from the `fetchpriority` option, which only controls network priority.
```

Comment:
Did you mean the fetch's request's [priority](https://fetch.spec.whatwg.org/#request-priority)?

Reply:
Ahh, thanks. It's `fetchpriority` for `<script>` and other resources, but not for `fetch()`. I cleaned that up and added links.